src/olympia/shelves/utils.py

```diff
@@ -0,0 +1,78 @@
+from urllib.parse import urlparse
+
+from django.conf import settings
+
+import requests
+from django_statsd.clients import statsd
+
+import olympia.core.logger
+
+
+log = olympia.core.logger.getLogger('z.shelves')
+
+
+def call_adzerk_server(placeholders):
+    """Call adzerk server to get sponsored addon results.
+
+    `placeholders` is a list of arbitrary strings that we pass so we can
+    identify the order of the results in the response dict."""
+    site_id = settings.ADZERK_SITE_ID
+    network_id = settings.ADZERK_NETWORK_ID
+    placements = [
+        {"divName": ph,
+         "networkId": network_id,
+         "siteId": site_id,
+         "adTypes": [5]} for ph in placeholders]
+
+    try:
+        log.info('Calling adzerk')
+        with statsd.timer('services.adzerk'):
+            response = requests.post(
+                settings.ADZERK_URL,
+                json={'placements': placements},
+                timeout=settings.ADZERK_TIMEOUT)
+        if response.status_code != 200:
+            raise requests.exceptions.RequestException()
+    except requests.exceptions.RequestException as e:
+        log.exception('Calling adzerk failed: %s', e)
+        statsd.incr('services.adzerk.fail')
+        return None
+    else:
+        statsd.incr('services.adzerk.success')
+    try:
+        return response.json()
+    except ValueError:
+        return {}
```

Comment:
Should we log in this case so we can track if/when this happens?

Reply:
https://github.com/mozilla/addons-server/pull/15670/commits/b06bd9ea4ba5671682a6637738e75fc48a569690